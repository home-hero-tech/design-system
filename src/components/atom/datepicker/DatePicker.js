import React from 'react';
import RDatePicker from 'react-datepicker';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import css from './DatePicker.module.scss';
import cssInput from '../input/Input.module.scss';

const DatePicker = ({ selected, format, onChange, dark, ...otherProps }) => {
  const className = dark ? 'c-datepicker--dark' : 'c-datepicker';
  const classes = classNames({
    [cssInput['c-input']]: true,
    [cssInput['c-input--dark']]: dark
  });

  return (
    <div className={css[className]}>
      <RDatePicker
        selected={selected}
        dateFormat={format}
        onChange={onChange}
        popperModifiers={{
          offset: {
            enabled: true
          },
          preventOverflow: {
            enabled: true,
            escapeWithReference: false,
            boundariesElement: 'viewport'
          }
        }}
        showPopperArrow={false}
        className={classes}
        calendarClassName={cssInput['c-calendar']}
        {...otherProps}
      />
    </div>
  );
};

DatePicker.propTypes = {
  format: PropTypes.string,
  onChange: PropTypes.func,
  selected: PropTypes.instanceOf(Date),
  dark: PropTypes.bool
};

DatePicker.defaultProps = {
  format: 'dd/MM/yyyy',
  onChange: f => f,
  selected: new Date(),
  dark: false
};

export default DatePicker;
