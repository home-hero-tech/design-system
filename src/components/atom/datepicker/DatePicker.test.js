import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DatePicker from './DatePicker';

describe('DatePicker', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(<DatePicker selected={new Date('2020-03-11T10:41:40.597Z')} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
