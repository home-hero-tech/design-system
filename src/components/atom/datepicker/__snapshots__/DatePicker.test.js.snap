// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should match with the snapshot 1`] = `
<div
  className="c-datepicker"
>
  <t
    allowSameDay={false}
    calendarClassName="c-calendar"
    className="c-input"
    customTimeInput={null}
    dateFormat="dd/MM/yyyy"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    enableTabLoop={true}
    excludeScrollbar={true}
    inlineFocusSelectedMonth={false}
    monthsShown={1}
    nextMonthButtonLabel="Next Month"
    nextYearButtonLabel="Next Year"
    onBlur={[Function]}
    onCalendarClose={[Function]}
    onCalendarOpen={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
      Object {
        "offset": Object {
          "enabled": true,
        },
        "preventOverflow": Object {
          "boundariesElement": "viewport",
          "enabled": true,
          "escapeWithReference": false,
        },
      }
    }
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    previousYearButtonLabel="Previous Year"
    readOnly={false}
    renderDayContents={[Function]}
    selected={2020-03-11T10:41:40.597Z}
    shouldCloseOnSelect={true}
    showMonthYearPicker={false}
    showPopperArrow={false}
    showPreviousMonths={false}
    showQuarterYearPicker={false}
    showTimeInput={false}
    showTimeSelect={false}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    withPortal={false}
  />
</div>
`;
